lib/silo/silo.rb

```diff
@@ -254,6 +254,13 @@ def set_metadata(data)
 
     def refresh(forced: false)
       self.class.check_prepared(@type)
+      unless dir_exists?("")
```

Comment:
Is this just attempting to start any connection with the silo?

Reply:
Yes, originally I used `get_silo` but the action that uses is extremely inefficient, whereas the directory checking action only uses one API call so it's considerably faster.

Reply:
It's actually something that could be worth looking into speeding up - if you try and use `get-silo` to locate a silo which doesn't exist, then the action will iterate through ALL existing accessible silos, download their metadata files, and read the name field to check for a match. In my testing, using `get-silo`, checking if 4 silos existed took over 30s.